packages/pyrtl/patches/ctypes_and_renderer.patch

```diff
@@ -0,0 +1,95 @@
+commit c5df13161c176b73c8071d1ae3bef51972ec0d97
+Author: Michael Christensen <chmdko@gmail.com>
+Date:   Sat Jun 5 00:25:52 2021 -0700
+
+    Move ctype import for Pyodide
+
+diff --git a/pyrtl/compilesim.py b/pyrtl/compilesim.py
+index 4ce0d45..403efc2 100644
+--- a/pyrtl/compilesim.py
++++ b/pyrtl/compilesim.py
+@@ -1,6 +1,5 @@
+ from __future__ import print_function, unicode_literals
+ 
+-import ctypes
+ import subprocess
+ import tempfile
+ import shutil
+@@ -8,7 +7,6 @@ import collections
+ from os import path
+ import platform
+ import sys
+-import _ctypes
+ 
+ from .core import working_block
+ from .wire import Input, Output, Const, WireVector, Register
+@@ -24,6 +22,11 @@ class DllMemInspector(collections.Mapping):
+     """ Dictionary-like access to a hashmap in a CompiledSimulation. """
+ 
+     def __init__(self, sim, mem):
++        try:
++            import ctypes
++        except ImportError:
++            raise PyrtlError('need ctypes installed')
```

Comment:
Do you really need to catch the error? Failing with the ImportError might be enough?

Reply:
Yes, I think it makes sense to fail with the ImportError in most of the cases except when trying to initialize the CompiledSimulation object; I think it's helpful to have at least one message giving the user an idea what do (i.e. use a different type of simulation).

I've updated the patch to just `import ctypes` where needed like you suggest, and now only display the additional PyrtlError in the object initialization.